### DISCUSSION TITLE
How to dynamically change the component value based on the value of other component?

### DISCUSSION THREAD

**Discussion by jessieweiyi** [data-driven-forms/react-forms#1066] [Need help] [answered]
I am having a use case that I need to change the selected value of a Select component based on the value of other component. 

I tried the resolveProps and it works on other props (like isDisabled, or even options) rather than value. 

In the [doc](https://data-driven-forms.org/schema/resolve-props), it mentions input.onChange can be used in this case. However, I cannot see input.onChange exists. 

I know that I can use the `set` in the condition, but my use case has some complicated logic that I would like to support in a function.

**Comment by rvsia** [comment, marked as answer]
Hello,

in this case, I would create a new component or enhance the current select.

1. Wrap it with `FormSpy` and set `subscription={{values: true}}` or set the form [subscription](https://data-driven-forms.org/components/renderer#subscription) for values. This ensures that the component will be always rendered when some value is changed.

2. Use `formOptions` to get change function and use this function in `useEffect` hook when you need it.

I have this component implemented in one of my forms: https://github.com/RedHatInsights/sources-ui/blob/master/src/components/AddApplication/AuthTypeSetter.js It changes authentication value according to selectedAuthentication.

**Reply by jessieweiyi** [reply]
Thanks it works perfectly
